paper.md

```diff
@@ -74,37 +82,81 @@ of the following classes:
 - continuous Galerkin methods
 - discontinuous Galerkin methods
 
-Since a discrete derivative operator is basically a linear operator, all of
+Since a discrete derivative operator is a linear operator, all of
 these SBP operators implement the basic interface of such linear operators
 (`AbstractMatrix` in Julia) such as multiplication by vectors and addition of
 operators. Finite difference and Fourier operators on periodic domains also
 allow the construction of rational functions of operators and their efficient
 implementation using the fast Fourier transform [@frigo2005design].
 
 In addition to basic SBP derivative operators, SummationByPartsOperators.jl
-contains a number of artificial dissipation and filtering operators, such as
+contains a number of related operators, such as
 
 - SBP artificial dissipation operators
 - spectral viscosity operators for Fourier methods
 - modal filter operators for Fourier methods and Legendre pseudospectral methods
 
+Using Julia's rich type system, all of these operators are implemented as their
+own types. This enables several optimizations such as a memory requirement
+independent of the number of grid points. In contrast, implementations based
+on sparse/banded matrices have a memory requirement growing linearly with the
+number of grid points. In addition, the structure of the operators can be taken
+into account for operator-vector multiplications, usually resulting in speed-ups
+of an order of magnitude or more on consumer hardware. For example, the application
+an optimized the sixth-order (in the interior) finite difference SBP operator
+[@almquist2017optimized] on a grid with 1000 nodes takes roughly 330 ns
+on a consumer CPU from 2017 (Intel® Core™ i7-8700K) using version v0.5.5 of
+SummationByPartsOperators.jl. In contrast, the same operation takes roughly
+3.9 microseconds using a sparse matrix format used in other implementations of
+this operator [@almquist2017optimized]. This benchmark is based on the following
+code, which also provides a very basic example of SummationByPartsOperators.jl.
+```julia
+# install the package if necessary
+using Pkg; Pkg.add("SummationByPartsOperators")
+
+# load the package
+using SummationByPartsOperators
+
+# create a finite difference SBP operator
+D = derivative_operator(MattssonAlmquistVanDerWeide2018Accurate(),
+                        derivative_order=1, accuracy_order=6,
+                        xmin=0.0, xmax=1.0, N=10^3)
```

Comment:
depending on the style on could reduce indentation here to just 4 characters.

But I think for readability the closing bracket should get its own line.

Reply:
Done in f158401

Reply:
This is the current output:
![image](https://user-images.githubusercontent.com/12693098/126197194-8f113238-c063-453b-99b0-0c848c87f7c1.png)

Reply:
Great. one could adda comma after `N=10^3` (then its easier to just add parameters writing in a new line), but that's a matter of taste.